core/shared/src/main/scala/scalaz/zio/internal/FiberContext.scala

```diff
@@ -145,36 +234,53 @@ private[zio] final class FiberContext[E, A](
                     case ZIO.Tags.Succeed =>
                       val io2 = nested.asInstanceOf[ZIO.Succeed[Any]]
 
-                      curIo = io.k(io2.value)
+                      if (traceExec && inTracingRegion) addTrace(k)
```

Comment:
That may be a silly question but what is the cost of all these checks when everything is disabled? Perhaps it is worth turning adding a nanopass AST phase to the compiler that is the identity function in `FiberContext` and one that calls `addTrace` in `FiberTracing` or something like that.

Reply:
It's written like that so that the JIT can compile it away completely when it sees that it will never be true because a `final` variable is false / was false for a long time. Benchmarks actually show an *increase* in performance with disabled compared to master because I turned inlining back on in sbt. I think anything involving closures might likely be slower than a conditional.

Reply:
`if` should be lot faster than a stackframe creation. Also we checked and there is no noticeable perf impact when tracing is off.

Reply:
Thanks both, that's very insightful 👍